autocorpus/autocorpus.py

```diff
@@ -80,6 +90,45 @@ def __get_sections(self, soup, config):
 
         return handle_not_tables(config["sections"], soup)
 
+    @staticmethod
+    def __load_pdf_models():
+        global pdf_converter
+        if pdf_converter is None:
+            pdf_converter = PdfConverter(
+                artifact_dict=create_model_dict(),
+            )
+
+    @staticmethod
+    def __extract_pdf_content(
+        file_path: Path,
+    ) -> bool:
+        """Extracts content from a PDF file.
+
+        Args:
+            file_path (Path): Path to the PDF file.
+
+        Returns:
+            bool: success status of the extraction process.
+        """
+        bioc_text, bioc_tables = None, None
+        global pdf_converter
+        Autocorpus.__load_pdf_models()
+        if pdf_converter:
+            # extract text from PDF
+            rendered = pdf_converter(str(file_path))
+            text, _, _ = text_from_rendered(rendered)
+            # seperate text and tables
+            text, tables = extract_table_from_pdf_text(text)
+            # format data for BioC
+            bioc_text = BioCTextConverter(text, "pdf", str(file_path))
+            bioc_text.output_bioc_json(file_path)
+            bioc_tables = BioCTableConverter(tables, str(file_path))
+            bioc_tables.output_tables_json(file_path)
```

Comment:
We shouldn't be outputting the files to disk during the processing like this. The way the html files are processed, they are saved to `self.main_text` and then saved to disk afterwards in `run.py`. We should use the same pattern for the pdf files for consistency. This will also solve the issue with the data files in the tests being modified when running the tests.